packages/cubejs-schema-compiler/src/adapter/PreAggregations.js

```diff
@@ -362,82 +363,175 @@ export class PreAggregations {
     }
   }
 
+  /**
+   * Returns function to determine whether pre-aggregation can be used or not
```

Comment:
Does this function need to return a predicate function, of could we simply call it directly, as in:

```
    static canUsePreAggregation(canUseQuery, refs, preagg): boolean {
        ...
    }
```

Alternatively, consider extracting a mini-class, especially since the code defines a ton of closures. Something along the lines of:

```
    class CanUsePreaggregation {
        constructor(private canUseQuery, private refs) {}

        canUse(preaggr) {
            ...
        }

        sortTimeDimensions(timeDimensions) {
           ...
        }
       
        <etc>
    }
```

Reply:
Yeah, it should return a predicate function that closures the initial query. The predicates will be executed with every pre-aggregation. As far as I know, a simple call is used only for unit testing. The root cause of this design - is performance.